pgjdbc/src/main/java/org/postgresql/plugin/AuthenticationPlugin.java

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright (c) 2021, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.postgresql.plugin;
+
+import org.postgresql.util.PSQLException;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+public interface AuthenticationPlugin {
+  @Nullable
+  String getPassword(AuthenticationRequestType type) throws PSQLException;
```

Comment:
Security best practices suggest that passwords should be used as `char[]` so they can be nullified after use: https://www.baeldung.com/java-storing-passwords